src/components/Reports/PartnerGivingAnalysisReport/PartnerGivingAnalysisReport.tsx

```diff
@@ -0,0 +1,412 @@
+import React, { useMemo, useState } from 'react';
+import { Box, CircularProgress, TablePagination } from '@material-ui/core';
+import { useTranslation } from 'react-i18next';
+import { AccountsListHeader as Header } from '../AccountsListLayout/Header/Header';
+// import { usePartnerGivingAnalysisReportQuery } from './GetPartnerGivingAnalysisReport.generated';
+import type { Order } from '../Reports.type';
+import { PartnerGivingAnalysisReportTable as Table } from './Table/Table';
+import { PartnerGivingAnalysisReportActions as Actions } from './Actions/Actions';
+import { Notification } from 'src/components/Notification/Notification';
+import { EmptyReport } from 'src/components/Reports/EmptyReport/EmptyReport';
+
+interface Props {
+  accountListId: string;
+  isNavListOpen: boolean;
+  onNavListToggle: () => void;
+  title: string;
+}
+
+export type Contact = {
+  giftAverage: number;
+  giftCount: number;
+  giftTotal: number;
+  currency: string;
+  lastGiftAmount: number;
+  lastGiftDate: string;
+  id: string;
+  name: string;
+  lifeTimeTotal: number;
+};
+
+type OrderBy = keyof Contact | null;
+
+const applyFilters = (contacts: Contact[], query: string): Contact[] => {
+  return contacts.filter((contact) => {
+    let matches = true;
```

Comment:
I think this function might be more readable if you just return true or false instead of storing that in a variable and returning it later. The `.forEach` could be converted to a `.some` or a `.every` and the return value could be used.

Although filter/sort/pagination eventually needs to be handled on the API so if this is temporary code, no need to spend more time on it.